lib/instrumental/agent.rb

```diff
@@ -371,10 +373,11 @@ def send_with_reply_timeout(message)
       end
     end
 
-    def open_socket(remote_host, remote_port, secure, verify_cert)
-      sock = TCPSocket.open(remote_host, remote_port)
+    def open_socket(sockaddr_in, secure, verify_cert)
+      sock = Socket.new(Socket::PF_INET, Socket::SOCK_STREAM, 0)
```

Comment:
Changed from `TCPSocket` to `Socket` so that we do DNS(?) resolution outside of here?